server/spec/one_sub_pool_per_stack_spec.rb

```diff
@@ -342,6 +342,9 @@
     # Simulate migration
     @host2_client.update_consumer({:guestIds => [{'guestId' => @guest_uuid}]})
 
+    # Need to wait a moment here for MySQL to catch up
+    sleep 2
```

Comment:
Nothing is jumping out at me from looking through this. There's nothing (intentionally) async going on here, and as such, there's nothing to really check. The lack of a return from that particular request doesn't help things any, either.

The issue itself stems from the two requests coming from different clients, and only occurs with MySQL. If the second request (list_entitlements) comes from the host2_client, the problem goes away there as well.